### PR TITLE
This improves the experimental feature template

### PR DESCRIPTION
**Pull request by sdake**
Reduce initial work for submission of an experimental feature
- Require an identified lead
- Require a short description of the feature

As RFC/design/documentation work is completed, check off and link each work product.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/experimental_feature.md [hunk 1]
```diff
@@ -1,31 +1,47 @@
...
+- [ ] Review of user stories in a work group meeting
+- [ ] RFC Authored
 - [ ] Feature is disabled by default
 - [ ] No impact on performance when the feature is disabled.
-- [ ] Steps for enabling the feature have been documented. This may include
-	instructions for building, running an `istioctl experimental command`, or
+- [ ] Instructions for enabling and using the feature have been documented. This may
+        include instructions for building, running an `istioctl experimental command`, or
 	using the preview profile. 
+- [ ] Feedback plan authoed. This may include user feedback meetings,
```
Comment (brian-avery) [inline .github/ISSUE_TEMPLATE/experimental_feature.md:33]: authored

File: .github/ISSUE_TEMPLATE/experimental_feature.md [hunk 2]
```diff
@@ -1,31 +1,47 @@
...
 
-This page lists the requirements for experimental features. Experimental features allow Istio work groups to collect more information
-on potential new functionality. This template should be completed before users try out any new experimental features. Please check off and 
-document the steps as they are completed.
+This page lists the requirements for an experimental feature. Completion of this template enables Istio work groups to collect
+information on potential new functionality. This template should be completed before users are exposed to any new
+experimental feature. Please complete this template during development.
 
-**Feature Name:** 
+This template shall be completed before users are exposed to any new experimetnal feature.
```
Comment (brian-avery) [inline .github/ISSUE_TEMPLATE/experimental_feature.md:12]: experimental

File: .github/ISSUE_TEMPLATE/experimental_feature.md [hunk 3]
```diff
@@ -1,31 +1,47 @@
...
 
 --- 
 
-### Requirements: 
+# Requirements:
 
-- [ ] Feature is reviewed in a work group meeting
-- [ ] RFC has been written
+- [ ] Review of user stories in a work group meeting
+- [ ] RFC Authored
```
Comment (brian-avery) [inline .github/ISSUE_TEMPLATE/experimental_feature.md:27]: Discussed this in Slack, but we have templates for design docs and RFCs in the Istio drive. We should link to them in the templates.

File: .github/ISSUE_TEMPLATE/experimental_feature.md [hunk 4]
```diff
@@ -1,31 +1,47 @@
...
+
+**User stories document:**
+
+**RFC document:**
+
+**Instructions document:**
+
+**Feedback plan:**
 
 > Link to instructions
```
Comment (brian-avery) [inline .github/ISSUE_TEMPLATE/experimental_feature.md:44]: These should probably be moved to their respective sections or removed

File: .github/ISSUE_TEMPLATE/experimental_feature.md [hunk 5]
```diff
@@ -1,31 +1,47 @@
...
+experimental feature. Please complete this template during development.
 
-**Feature Name:** 
+This template shall be completed before users are exposed to any new experimetnal feature.
+
+# Mandatory fields for submission
+
+**Feature name:**
+
+**Primary lead:**
```
Comment (howardjohn) [inline .github/ISSUE_TEMPLATE/experimental_feature.md:18]:
```suggestion
**Primary lead(s):**
```

may be more inclusive, many have multiple leads

Comment (sdake) [reply .github/ISSUE_TEMPLATE/experimental_feature.md:18]: agreed, added.

File: .github/ISSUE_TEMPLATE/experimental_feature.md (hunk 4 above)
Comment (sdake) [reply .github/ISSUE_TEMPLATE/experimental_feature.md:44]: agreed - good suggestion. applied.

File: .github/ISSUE_TEMPLATE/experimental_feature.md (hunk 3 above)
Comment (sdake) [reply .github/ISSUE_TEMPLATE/experimental_feature.md:27]: agreed applied.

File: .github/ISSUE_TEMPLATE/experimental_feature.md (hunk 2 above)
Comment (sdake) [reply .github/ISSUE_TEMPLATE/experimental_feature.md:12]: resolved.

File: .github/ISSUE_TEMPLATE/experimental_feature.md (hunk 1 above)
Comment (sdake) [reply .github/ISSUE_TEMPLATE/experimental_feature.md:33]: clarified.

**Comment by sdake** [comment]
rendering: https://github.com/sdake/enhancements/blob/master/.github/ISSUE_TEMPLATE/experimental_feature.md

**Comment by sdake** [comment]
Seeing the rendering, I believe there are more improvements that can be made to increase brevity (which can improve velocity).

Also, I am not sure how I feel about the links being directly integrated into the checkboxes. This will require individuals to edit the templates with the checkbox links, which may end in doom for some :) @brian-avery any thought on this (this integration was your request).

Finally, and perhaps this is the biggest improvement to be made to all these templates, is adding comments for each field of the template, such that the person filling out the template knows what to do with the template. This is (one) way to improve the brevity of the template.

**Comment by brian-avery** [comment]
Yeah, I kind of liked them separate. My comment was more along the lines that we had:

[  ]  Steps for enabling the feature have been documented. This may include instructions for building, running an istioctl experimental command, or using the preview profile.
[  ]  Plan exists for getting feedback. This may include user feedback meetings, discuss.istio.io conversations, GitHub issues, or mailing lists.

>    Link to instructions
>   Plan for feedback

And I was just proposing keeping the boxes that we had for instructions and feedback with the check boxes for the respective content.

**Comment by brian-avery** [comment]
By the way, that same rendering can be seen directly from the PR.

**Comment by sdake** [comment]
@brian-avery I don't believe I have changed the sprit nor intent of the original template. I did add a few fields which I felt were required and dropped the comments into the document such that they don't render.

Cheers,
-steve

**Comment by sdake** [comment]
> Yeah, I kind of liked them separate. My comment was more along the lines that we had:
> 
> [ ] Steps for enabling the feature have been documented. This may include instructions for building, running an istioctl experimental command, or using the preview profile.
> [ ] Plan exists for getting feedback. This may include user feedback meetings, discuss.istio.io conversations, GitHub issues, or mailing lists.
> 
> > Link to instructions
> 
> > Plan for feedback
> 
> And I was just proposing keeping the boxes that we had for instructions and feedback with the check boxes for the respective content.

@brian-avery - ok - makes sense. I did take a stab at documenting the template for the development community, as they feel like the primary audience. The comments (should) be hidden during rendering, but not during form fillout or editng. Although a merge would be required to experiment 👍

Cheers,
-steve

File: .github/ISSUE_TEMPLATE/experimental_feature.md [hunk 6]
```diff
@@ -1,31 +1,64 @@
...
 
-# Experimental Feature Requirements
+[//]: # (The syntax preceeding this line is a comment marker used to help guide the issue author with submission of an issue)
+[//]: # (to github. Unlike HTML comments commonly used throughout istio.io documentation, this comment will not be rendered)
+[//]: # (by github. Comments must be separated by carriage return preceding and concluding the text and be a single line.)
 
-This page lists the requirements for experimental features. Experimental features allow Istio work groups to collect more information
-on potential new functionality. This template should be completed before users try out any new experimental features. Please check off and 
-document the steps as they are completed.
+[//]: # (This page lists the requirements for an experimental feature. Completion of this template enables Istio work groups)
```
Comment (brian-avery) [inline .github/ISSUE_TEMPLATE/experimental_feature.md:10]: Cool. I wasn't aware of these.

File: .github/ISSUE_TEMPLATE/experimental_feature.md [hunk 7]
```diff
@@ -1,31 +1,64 @@
...
+
+- [ ] [Documention](insert_your_link_here) for enabling and using the feature.
+
+[//]: # (The documentation instructions may exist on the developer wiki or the team drive. They may include instructions)
+[//]: # (for building running a `istioctl experimental command`, or using the preview profile,)
+[//]: # (or any other relevant information.)
+
+- [ ] [Feedback plan](insert_your_link_here).
+
+[//]: # (This may include user feedback meetings, discuss.istio.io conversations, GitHub issues, or mailing lists.)
```
Comment (brian-avery) [inline .github/ISSUE_TEMPLATE/experimental_feature.md:60]: Thanks for adding all of this documentation. This is awesome.

File: .github/ISSUE_TEMPLATE/experimental_feature.md [hunk 8]
```diff
@@ -1,31 +1,64 @@
...
+[//]: # (into the RFCs, which do need approval for alpha. You may find value to negotiate within the work group where the)
+[//]: # (user stories are presented to help clarify the user stories.)
 
-> Plan for feedback
+- [ ] [RFC Authored] - [create an RFC using template](https://docs.google.com/document/d/1ewJoCcw5-04crH-M0xw4zFxz1cfwVCPnNyW4K3m4Yyc/template/preview).
+
+[//]: # (An RFC is mandatory to graduate to experimental. The RFC does not have to be reviewed in a work group)
+[//]: # (meeting to graduate to experimental.)
+
+- [ ] [Documention](insert_your_link_here) for enabling and using the feature.
```
Comment (brian-avery) [inline .github/ISSUE_TEMPLATE/experimental_feature.md:52]: Documentation

**Comment by linsun** [comment]
I think the newly addition to the template could add a lot more work from developers and many info are already in RFC.  I'll let others to chime in.  Also, if we require this level of details in the experimental template, we need to update alpha/beta/stable template as well.

**Review by brian-avery — APPROVED** [review]

**Comment by brian-avery** [comment]
@linsun I don't think anything was actually added. @sdake rephrased the content and added comments explaining what each field was asking for.

**Comment by sdake** [comment]
New feature request can be viewed here: https://github.com/sdake/enhancements/issues/new?template=experimental_feature.md

**Comment by sdake** [comment]
@brian-avery to be fair, I did add two fields:
1. the leads field
1. the short description field

everything else is just reword / reorganization / comments.

File: .github/ISSUE_TEMPLATE/experimental_feature.md (hunk 8 above)
Comment (sdake) [reply .github/ISSUE_TEMPLATE/experimental_feature.md]: resolved.

**Comment by sdake** [comment]
rendering of the page on github: https://github.com/sdake/enhancements/blob/master/.github/ISSUE_TEMPLATE/experimental_feature.md

**Comment by linsun** [comment]
sorry - my bad @sdake @brian-avery , i had thought it is adding a lot of new things initially but realized it is more adding clarifications.  Yes, updating 4 templates would be nice.

**Comment by linsun** [comment]
> rendering of the page on github: https://github.com/sdake/enhancements/blob/master/.github/ISSUE_TEMPLATE/experimental_feature.md

rendering LG!  hard to tell from just reviewing the lines :)

**Comment by sdake** [comment]
@linsun would you mind approving this - I will follow up with comments for the others after the 20th. I want to submit an experimental feature for the ingress gateway changes.

Thanks,
-steve

**Review by linsun — APPROVED** [review]

**Pull request merged**